ted/src/main/java/frc/robot/subsystems/PowerDistributionPanelWatcherSubsystem.java

```diff
@@ -82,4 +105,27 @@ public void periodic() {
             SmartDashboard.putNumber(nextSpy.getActionDescription(), current);
         }
     }
+
+    /**
+     * A method to check for and handle brownout events
+     * runs the brownoutAction once brownoutEventsBeforeAction brownouts are detected.
+     */
+    private void handleBrownouts(){
+        if (isBrownedOut()){
+            brownoutEventCount += 1;
+            if (this.handleBrownouts && (brownoutEventCount >= this.brownoutEventsBeforeAction)){
+                brownoutAction.run();
```

Comment:
Seems like to me that runnables can/are long-running (seems like most times run in a forked thread).  Here we rely on them to be short-lived.